misk-hibernate/src/main/kotlin/misk/hibernate/DbChild.kt

```diff
@@ -0,0 +1,10 @@
+package misk.hibernate
+
+/**
+ * Marker interface for persistent entities. Ensures that only persistent entities can be passed
+ * into [Session] methods.
+ */
+interface DbChild<R : DbRoot<R>, T : DbChild<R, T>> : DbMember<R, T> {
```

Comment:
should we rename to DbEntity ?

Reply:
This is specifically a child entity, no? So `DbChild` seems appropriate. `DbMember` is the type that I think could be renamed to `DbEntity`, since it covers both root entities and child entities.

Reply:
I'm thinking:
`DbMember` -> `DbSharded`
`DbEntity` -> `DbUnsharded`

I want to name and shame anything that is not sharded by calling this out with the hope that this does not become the default choice.

I'd like to keep this names short and sweet because they're going to be everywhere. It seems an `Entity` suffix doesn't add anything above what the `Db` prefix already says.

Reply:
It's a bit awkward though because that means `DbSharded` will extend `DbUnsharded`... and all Session APIs will accept a `DbUnsharded` parameter.

I don't think we can have the base class of all persistent entities be called `DbUnsharded`.

Instead I will try this:
* Add a `DbUnsharded` subclass to `DbEntity`.
* In `Session::save` method ensure enforce that an entity either subclasses `DbUnsharded` or `DbSharded`.

Not perfect but we will iterate on this.